### PR TITLE
feature: Show message box only within zones

### PR DESCRIPTION
**Pull request by jorelosorio**
Closes https://github.com/AtB-AS/kundevendt/issues/4061

Quick resume: The city box is only shown when the origin and destination are in the same city zone.

### CODE REVIEW

File: src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_DashboardStack/Dashboard_TripSearchScreen/components/CityZoneMessage.tsx [hunk 1]
```diff
@@ -49,11 +55,13 @@ export const CityZoneMessage: React.FC<CityZoneMessageProps> = ({
...
-    text: cityZone.name,
-    onPress: () => openUrlForCityZone(cityZone),
-  }));
+  const messageActions = [
+    {
+      id: `${fromCityZone.id}_action`,
+      text: fromCityZone.name,
+      onPress: () => openUrlForCityZone(fromCityZone),
+    },
+  ];
```
Comment (rosvik) [inline src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_DashboardStack/Dashboard_TripSearchScreen/components/CityZoneMessage.tsx:64]: Will there ever be more than one messageAction now? If so, maybe we could simplify the logic here and in `CityZoneBox`.

File: src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_DashboardStack/Dashboard_TripSearchScreen/components/CityZoneMessage.tsx [hunk 2]
```diff
@@ -49,11 +55,13 @@ export const CityZoneMessage: React.FC<CityZoneMessageProps> = ({
...
 
   if (messageActions) {
```
[the remark this answers is not in the document]
Comment (jorelosorio) [reply src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_DashboardStack/Dashboard_TripSearchScreen/components/CityZoneMessage.tsx]: Fixed :)

File: src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_DashboardStack/Dashboard_TripSearchScreen/components/CityZoneMessage.tsx (hunk 1 above)
Comment (jorelosorio) [reply src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_DashboardStack/Dashboard_TripSearchScreen/components/CityZoneMessage.tsx:64]: Not now, but I am working another ticket that requires more than one message action

**Review by gorandalum — COMMENTED** [review]
The check on the `enabled` flag, is that no longer necessary?

**Comment by jorelosorio** [comment]
> The check on the `enabled` flag, is that no longer necessary?

Yes it should be used, but is not included in the condition I missed it, I am working on a related task where I hope include it again using:

```
if (!fromCityZone?.enabled || !toCityZone?.enabled) {
    return null;
}
```

**Pull request merged**